crates/fj-app/src/main.rs

```diff
@@ -45,18 +45,29 @@ fn main() -> anyhow::Result<()> {
 
     let args = Args::parse();
     let config = Config::load()?;
-
     let path = config.default_path.unwrap_or_else(|| PathBuf::from(""));
     let parameters = args.parameters.unwrap_or_else(Parameters::empty);
     let shape_processor = ShapeProcessor {
         tolerance: args.tolerance,
     };
 
-    let model = if let Some(model) = args.model.or(config.default_model) {
+    let path_of_model = path.canonicalize().unwrap_or_default();
+
+    let model = if let Some(model) =
+        args.model.or(config.default_model).as_ref()
+    {
         let mut model_path = path;
         model_path.push(model);
         Some(Model::from_path(model_path.clone()).with_context(|| {
-            format!("Failed to load model: {}", model_path.display())
+            if path_of_model.as_os_str().is_empty() {
```

Comment:
Under which circumstances can `path_of_model` be empty? I suspect this can only happen, if `path.canonicalize()` above returns an `Err`.

If that is true, we can make things a bit more direct, doing something like this:

```
let abs_path = if let Ok(path_of_model) = path.canonicalize() {
    // insert error message using `path_of_model` here
} else {
    // insert error message that doesn't use `path_of_model` here
}
```

As a side note, this would be a prime use case [`let-else`](https://rust-lang.github.io/rfcs/3137-let-else.html). This feature isn't stable yet, but should become available soon.

Reply:
@hannobraun I will truly do another PR again,
I also tried to use that `if let` case but I was running it with `Some` and couldn't figure it out, and settle with using the simple `if else`.

Reply:
> I will truly do another PR again,

Thank you, that's great!

---

In case you're wondering how to figure out these kinds of things, here's where to find the necessary information in the standard library API reference:

If you check out the documentation, you can see that [`Path::canonicalize`](https://doc.rust-lang.org/std/path/struct.Path.html#method.canonicalize) returns a `Result` (specifically an [`io::Result`](https://doc.rust-lang.org/std/io/type.Result.html), which is defined in terms of [`result::Result`](https://doc.rust-lang.org/std/result/enum.Result.html)). `Result` has `Ok` and `Err` variants, which you can use with `if let` or `match`.

`Some`, on the other hand, is a variant of [`Option`](https://doc.rust-lang.org/std/option/enum.Option.html), which is *not* returned by `canonicalize`.